src/main/host/syscall/handler/socket.rs

```diff
@@ -127,13 +126,9 @@ impl SyscallHandler {
             let desc_table = ctx.objs.process.descriptor_table_borrow();
             let desc = Self::get_descriptor(&desc_table, fd)?;
 
-            let file = match desc.file() {
-                CompatFile::New(file) => file,
-                // if it's a legacy file, use the C syscall handler instead
-                CompatFile::Legacy(_) => {
-                    drop(desc_table);
-                    return Self::legacy_syscall(c::syscallhandler_bind, ctx);
-                }
+            let CompatFile::New(file) = desc.file() else {
+                // we don't have any C socket objects
```

Comment:
Do we think these are now unreachable? Maybe `panic` or `debug_panic`?

Reply:
These can be reached if the syscall is called on a non-socket C file, for example `connect()` on a regular file. I originally did have a panic here, but luckily the socket tests caught it.